src/main/java/com/esri/samples/editing/edit_and_sync_features/EditAndSyncFeaturesSample.java

```diff
@@ -0,0 +1,403 @@
+          // on secondary mouse click, clear feature selection
+          for (Layer layer : mapView.getMap().getOperationalLayers()) {
+            final FeatureLayer featureLayer = (FeatureLayer) layer;
+            featureLayer.clearSelection();
+          }
+        }
+      });
+
+      // add map view, controlsVBox and progressBar to stack pane
+      stackPane.getChildren().addAll(mapView, controlsVBox, progressIndicator);
+      StackPane.setAlignment(controlsVBox, Pos.TOP_LEFT);
+      StackPane.setMargin(controlsVBox, new Insets(10, 0, 0, 10));
+      StackPane.setAlignment(progressIndicator, Pos.BOTTOM_CENTER);
+      StackPane.setMargin(progressIndicator, new Insets(0, 0, 50, 0));
+
+    } catch (Exception e) {
+      // on any error, display the stack trace
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Generates a local geodatabase and sets it to the map.
+   *
+   * @args extent     the extent of the map from which a geodatabase is generated
+   */
+  public void generateGeodatabase(Envelope extent) {
+
+    // define geodatabase sync task
+    geodatabaseSyncTask = new GeodatabaseSyncTask("https://sampleserver6.arcgisonline.com/arcgis/rest/services/Sync/WildfireSync/FeatureServer");
+    geodatabaseSyncTask.loadAsync();
+    geodatabaseSyncTask.addDoneLoadingListener(() -> {
+
+      // create generate geodatabase parameters for the current extent
+      final ListenableFuture<GenerateGeodatabaseParameters> defaultParameters = geodatabaseSyncTask
+              .createDefaultGenerateGeodatabaseParametersAsync(extent);
+      defaultParameters.addDoneListener(() -> {
+        try {
+          // set parameters and don't include attachments
+          GenerateGeodatabaseParameters parameters = defaultParameters.get();
+          parameters.setReturnAttachments(false);
+
+          // create a temporary file for the geodatabase
+          File tempFile = File.createTempFile("gdb" + replica.getAndIncrement(), ".geodatabase");
+          tempFile.deleteOnExit();
+
+          // create and start the job
+          GenerateGeodatabaseJob geodatabaseJob = geodatabaseSyncTask.generateGeodatabase(parameters, tempFile.getAbsolutePath());
+          geodatabaseJob.start();
+
+          // show progress
+          showProgress(geodatabaseJob);
+
+          // get geodatabase when done
+          geodatabaseJob.addJobDoneListener(() -> {
+            if (geodatabaseJob.getStatus() == Job.Status.SUCCEEDED) {
+              geodatabase = geodatabaseJob.getResult();
+              geodatabase.loadAsync();
+              geodatabase.addDoneLoadingListener(() -> {
+                if (geodatabase.getLoadStatus() == LoadStatus.LOADED) {
+                  // add the geodatabase FeatureTables to the map as a FeatureLayer
+                  geodatabase.getGeodatabaseFeatureTables().forEach(geodatabaseFeatureTable -> {
+                    // TODO: why is the adding of the FT to the map not within a doneListener?
```

Comment:
another TODO to be removed. The adding of the FT to the map is within a done loading listener for the geodatabase?